contracts/MintShop1155.sol

```diff
@@ -678,14 +679,14 @@ contract MintShop1155 is Sweepable, ReentrancyGuard {
   */
   function getWhitelistStatus(uint256[] calldata _ids,
     address[] calldata _addresses) external view returns (bool[][] memory) {
-    bool[][] memory whitelistStatus;
-    for (uint256 i = 0; i < _ids.length; i++) {
-      uint256 id = _ids[i];
-      uint256 whitelistVersion = whitelists[id].currentWhitelistVersion;
-      for (uint256 j = 0; j < _addresses.length; j++) {
-        bytes32 addressKey = keccak256(abi.encode(whitelistVersion,
-          _addresses[j]));
-        whitelistStatus[j][i] = whitelists[id].addresses[addressKey];
+    bool[][] memory whitelistStatus = new bool[][](_addresses.length);
+    for (uint256 i = 0; i < _addresses.length; i++) {
+      whitelistStatus[i] = new bool[](_ids.length);
+      for (uint256 j = 0; j < _ids.length; j++) {
+        uint256 id = _ids[j];
+        uint256 whitelistVersion = whitelists[id].currentWhitelistVersion;
+        bytes32 addressKey = keccak256(abi.encode(whitelistVersion, _addresses[i]));
+        whitelistStatus[i][j] = whitelists[id].addresses[addressKey];
```

Comment:
If we're changing the order of the indexing we also need to change the order of the `_ids` and `_addresses` parameters.